src/test/java/BaseTest.java

```diff
@@ -1,18 +1,193 @@
+        submitButton.click();
+    }
+
+
+    //Methods for playing a song
+    public void enterAllSongs() {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(4));
+        WebElement allSongs = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("a[href='#!/songs']")));
+        allSongs.click();
+
+        // WebElement allSongs = driver.findElement(By.cssSelector("a[href='#!/songs']"));
+        //allSongs.click();
+
+        //Thread.sleep(1000);
+        // allSongs.click();
+        //
     }
 
+    public void selectSong() {
+        WebElement song = driver.findElement(By.xpath("//tr[@class='song-item']"));
+        song.click();
+
+        //throws InterruptedException
+        //WebElement song = driver.findElement(By.xpath("//tr[@class='song-item']"));
+        //Thread.sleep(1000);
+
+    }
+
+    public void enterButtonPlaySong() {
+        WebElement buttonPlaySong = driver.findElement(By.xpath("//span[@title='Play or resume']"));
+        Actions actions = new Actions(driver);
+        actions.click(buttonPlaySong).perform();
+    }
+
+
+    public boolean isDisplayedPlayingSong() {
+
+        WebElement songIsPlaying = driver.findElement(By.cssSelector("[data-testid = 'sound-bar-play']"));
+        return songIsPlaying.isDisplayed();
+
+
+    }
+
+    //Methods to rename a playlist
+    public void doubleClickOnPlaylist() {
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(1));
+
+        Actions actions = new Actions(driver);
+        WebElement myPlayList = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("/html/body/div/div/div/nav/section[2]/ul/li[3]/a")));
+        actions.doubleClick(myPlayList).perform();
+
+    }
+
+    public void provideNewName() {
+
+
+        WebElement nameField = driver.findElement(By.xpath("/html/body/div/div/div/nav/section[2]/ul/li[3]/a"));
+        nameField.sendKeys((Keys.chord(Keys.CONTROL, "a", Keys.BACK_SPACE)));
+        nameField.sendKeys(playlistName);
+        nameField.sendKeys(Keys.ENTER);
+    }
+
+    public boolean doesPlaylistExist() {
+        WebElement playlistElement = driver.findElement(By.xpath("//a[text()='" + playlistName + "']"));
+        return playlistElement.isDisplayed();
+    }
+
+
+    //Additional things for Login test
+    public boolean isHomePageElementDisplayed() {
+        WebElement homePage = driver.findElement(By.xpath("/html/body/div/div/div/nav/section[1]/ul/li[1]/a"));
+        return homePage.isDisplayed();
+
+    }
+
+    public void logIn() {
+        provideEmail("janezelenova@gmail.com");
+        providePassword("Floridaliving2023$");
+
+        //AND user clicks a submit button
+
+        clickSubmit();
+    }
+
+    @DataProvider(name = "incorrectLoginData")
+    public Object[][] getDataProviders() {
+        return new Object[][]{{"invalid@gmail.com", "invalidPass"}, {"onlyEmail@gmail.com", ""}, {"", ""}
+
+        };
+    }
```

Comment:
The code is not compiled! Please finish the homework!You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/226535739-65a57571-1b24-48b1-9853-d27d79aa3777.png)